src/lighteval/models/base_model.py

```diff
@@ -354,9 +354,17 @@ def greedy_until(
             position=0,
             disable=self.disable_tqdm,
         ):
-            # Longest context in the current split is the first item (since we sort reversed)
-            longest_context_continuation_size_in_split = len(dataset[0].tokenized_context) + dataset[0].generation_size
-            max_context_continuation_size_allowed = min(longest_context_continuation_size_in_split, self.max_length)
+            if dataset[0].generation_size is None:
+                # No constraints on the generation size: max length allowed is the max model context
+                max_context_continuation_size_allowed = self.max_length
```

Comment:
`self.max_length` is the total max length of the model, that is: context + generation.
I think we need to set it to max_length - max_context_length